src/app/upgrade.cpp

```diff
@@ -233,8 +233,8 @@ namespace
     {
         struct KeyMapping
         {
-            QString oldKey;
             QString newKey;
+            QString oldKey;
```

Comment:
I think this is wrong fix.
See its original usage: https://github.com/qbittorrent/qBittorrent/commit/39f054eef628b199c5c3d2d37fa62dd98a4a48f2#diff-727081584f853e38da2cfa7a916822653a889c2ce33b39650123afbbf72e9acbL149
It maps from left string to right string.

I suspect the problem you are seeing stems from the (affect) setting doesn't read from the new key but instead from the old. For example, `State/BannedIPs` is used in actual code but here it is being migrated away.

Reply:
> I think this is wrong fix.
> See its original usage: [39f054e#diff-727081584f853e38da2cfa7a916822653a889c2ce33b39650123afbbf72e9acbL149](https://github.com/qbittorrent/qBittorrent/commit/39f054eef628b199c5c3d2d37fa62dd98a4a48f2#diff-727081584f853e38da2cfa7a916822653a889c2ce33b39650123afbbf72e9acbL149)
> It maps from left string to right string.

So why this fix is incorrect? Left string is NEW setting name. To migrate we should read OLD setting (using right string) and write it as new one so then code can access it without any runtime mapping.

Reply:
>Left string is NEW setting name. To migrate we should read OLD setting (using right string) and write it as new one so then code can access it without any runtime mapping.

Then how do you explain `{"AddNewTorrentDialog/TreeHeaderState", "AddNewTorrentDialog/qt5/treeHeaderState"},`?
Clearly the qt5 one is the new one, isn't?

Reply:
> Then how do you explain `{"AddNewTorrentDialog/TreeHeaderState", "AddNewTorrentDialog/qt5/treeHeaderState"},`?
> Clearly the qt5 one is the new one, isn't?

New name is used in the code:
https://github.com/qbittorrent/qBittorrent/blob/bdcb00a3b2c4d34d14261ebaa5a89a20a8f52318/src/gui/addnewtorrentdialog.cpp#L178

Reply:
>See its original usage: 39f054e#diff-727081584f853e38da2cfa7a916822653a889c2ce33b39650123afbbf72e9acbL149

If I understand correctly the usage is different now.
Earlier: It loaded the "true" key stored in config from a pseudo key.
Now: It loads the value from the old true key and stores the pseudo key as the new true key along assigning to it the value of the old true key. Then removes the old true key from config.

Reply:
>New name is used in the code:

That is the old key IMO. Also the (deleted) runtime mapping will map it to new key, isn't? And in settings file it will use the new key, right?

Reply:
>New name is used in the code:

Wow. I think this should be considered a mistake. I think we need versioned strings for saving widget states between qt major versions (qt5/qt6). They might not be compatible between them.

Reply:
>Earlier: It loaded the "true" key stored in config from a pseudo key.
Now: It loads the value from the old true key and stores the pseudo key as the new true key, assigning to it the value of the old true key. Then removes the old true key from config.

I did a shitty explanation. Let me re-try:
**Old:** Code uses everywhere the left value, the mapping translated it to the right value, and the right value was stored/loaded from config.
**New:** Migration code: Basically wants to drop the runtime translation, and store the new key directly to config. So left key == new key, right key == old key. Same as before.

Reply:
> new key (right side).

Right side is OLD key.

Reply:
> Wow. I think this should be considered a mistake. I think we need versioned strings for saving widget states between qt major versions (qt5/qt6). They might not be compatible between them.

We start to do it in a different manner since Qt6:
https://github.com/qbittorrent/qBittorrent/blob/bdcb00a3b2c4d34d14261ebaa5a89a20a8f52318/src/gui/addnewtorrentdialog.cpp#L174-177